Xamarin.Essentials/DeviceDisplay/DeviceDisplay.win32.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace Xamarin.Essentials
+{
+    public static partial class DeviceDisplay
+    {
+        const float mmToInch = 0.03937008f;
+
+        static bool keepingScreenOn = false;
+
+        static bool PlatformKeepScreenOn
+        {
+            get => keepingScreenOn;
+            set
+            {
+                keepingScreenOn = value;
+                NativeMethods.SetThreadExecutionState((value ? NativeMethods.EXECUTION_STATE.DISPLAY_REQUIRED : 0) | NativeMethods.EXECUTION_STATE.CONTINUOUS);
+            }
+        }
+
+        static DisplayInfo GetMainDisplayInfo()
+        {
+            var hMonitor = NativeMethods.MonitorFromWindow(NativeMethods.GetDesktopWindow(), NativeMethods.MONITOR_DEFAULTTO.PRIMARY);
+            if (hMonitor != IntPtr.Zero)
+            {
+                var mi = default(NativeMethods.MONITORINFOEX);
+                mi.cbSize = Marshal.SizeOf(mi);
+
+                if (NativeMethods.GetMonitorInfo(hMonitor, ref mi))
+                {
+                    var dc = NativeMethods.CreateDC(mi.szDevice, null, null, IntPtr.Zero);
+                    if (dc != IntPtr.Zero)
+                    {
+                        var xpixels = NativeMethods.GetDeviceCaps(dc, NativeMethods.DeviceCap.HORZRES);
+                        var xsize = NativeMethods.GetDeviceCaps(dc, NativeMethods.DeviceCap.HORZSIZE);
+                        var rawDpiX = xpixels / (xsize * mmToInch);
+
+                        var ypixels = NativeMethods.GetDeviceCaps(dc, NativeMethods.DeviceCap.VERTRES);
+
+                        var orientation = xpixels > ypixels ? DisplayOrientation.Landscape : DisplayOrientation.Portrait;
+                        NativeMethods.DeleteDC(dc);
+
+                        return new DisplayInfo(xpixels, ypixels, rawDpiX, orientation, DisplayRotation.Unknown);
+                    }
+                }
+            }
+
+            return default;
+        }
+
+        static void StartScreenMetricsListeners()
+        {
+        }
+
+        static void StopScreenMetricsListeners()
+        {
+        }
+
+        static class NativeMethods
+        {
+            [DllImport("Kernel32", SetLastError = true)]
+            internal static extern EXECUTION_STATE SetThreadExecutionState(EXECUTION_STATE esFlags);
+
+            internal enum EXECUTION_STATE : uint
+            {
+                CONTINUOUS = 0x80000000,
+                DISPLAY_REQUIRED = 0x00000002,
+            }
+
+            [DllImport("user32")]
+            internal static extern IntPtr GetDesktopWindow();
+
+            [DllImport("user32")]
+            internal static extern IntPtr MonitorFromWindow(IntPtr hwnd, MONITOR_DEFAULTTO dwFlags);
+
+            internal enum MONITOR_DEFAULTTO : int
+            {
+                NULL = 0,
+                PRIMARY = 1,
+                NEAREST = 2,
+            }
+
+            [DllImport("user32")]
```

Comment:
Add ExactSpelling, switch to GetMonitorInfoW for less string marshalling overhead